### PR TITLE
config/wayland: add page

### PR DESCRIPTION
**Pull request by ericonr**
Add a Wayland page, loosely based on what's available on the wiki. I can add/remove/correct information according to suggestions and opinions here.

### CODE REVIEW

**Comment by ericonr** [comment]
A little update based on feedback from IRC.

File: src/config/wayland.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Wayland
+
+This section details the manual installation and configuration of Wayland
+compositors and related services and utilities.
+
+## Installation
+
+Unlike [Xorg](./xorg/index.md), Wayland isn't implemented as a server that
+applications like a window manager and compositor talk to. Instead, the Wayland
+compositor implements all functions related to displaying windows on the screen.
```
Comment (st3r4g) [inline src/config/wayland.md]: I'd reword this to make it clearer:
```suggestion
Unlike [Xorg](./xorg/index.md), Wayland implementations combine the display
server, the window manager and the compositor in a single application.
```

File: src/config/wayland.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+```
+if test -z "${XDG_RUNTIME_DIR}"; then
+    export XDG_RUNTIME_DIR=/tmp/${UID}-runtime-dir
+    if ! test -d "${XDG_RUNTIME_DIR}"; then
+        mkdir -p "${XDG_RUNTIME_DIR}"
+        chmod 0700 "${XDG_RUNTIME_DIR}"
+    fi
+fi
+```
```
Comment (st3r4g) [inline src/config/wayland.md]: Although it appears [here](https://wayland.freedesktop.org/building.html), @Duncaen pointed out that this snippet has some flaws, but I don't remember the details now.

File: src/config/wayland.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+compositors, and is installed as a dependency for most of them. Its package is
+`xorg-server-xwayland`. For Weston, the correct package is `weston-xwayland`.
+
+## Configuration
+
+The Wayland API uses the `XDG_RUNTIME_DIR` environment variable to determine the
+directory the Wayland socket. If this environment isn't already set in your
+setup, you can add this snippet to your shell profile or to a script that
+starts-up your Wayland compositor of choice. This shouldn't be necessary with
+any of the desktop environments.
```
Comment (st3r4g) [inline src/config/wayland.md]: You could mention that `elogind` sets this up for you.

File: src/config/wayland.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        mkdir -p "${XDG_RUNTIME_DIR}"
+        chmod 0700 "${XDG_RUNTIME_DIR}"
+    fi
+fi
+```
+
+It is also possible that some applications use the `XDG_SESSION_TYPE`
+environment variable in some way, which requires that you set it to `wayland`.
+
+## Video drivers
```
Comment (st3r4g) [inline src/config/wayland.md]: This part is important, I wouldn't put it at the end. Having `sway` crash because you forgot the drivers is a common mistake, and not immediately recognized by everybody. Ideally, the drivers would be pulled in as a dependency as soon as you install a GL application (Wayland compositor), but this requires `mesa` to be a dependency of `libglvnd` which is not trivial at the moment.

File: src/config/wayland.md (hunk 1 above)
Comment (ericonr) [reply src/config/wayland.md]: That does sound better. Thanks!

File: src/config/wayland.md (hunk 2 above)
Comment (ericonr) [reply src/config/wayland.md]: As you mentioned below, elogind also sets it up, so it's definitely a solution. However, I feel like including the script could make life easier for some people (as it did for me, as a matter of fact). If it has issues, though, it's probably a good idea to remove it, seeing as we want the Void documentation to be properly curated.

Comment (ericonr) [reply src/config/wayland.md]: I'm also thinking of including the `XDG_SESSION_TYPE` variable, and the fact that you're supposed to set it to `wayland`. Do you know any applications that use its value? My own scripts do, and I feel like it's the "correct" thing to do, but I have no idea if it's necessary.

File: src/config/wayland.md (hunk 4 above)
Comment (ericonr) [reply src/config/wayland.md]: > this requires mesa to be a dependency of libglvnd which is not trivial at the moment.

Is there any help I can give in that front? Would it be a packaging issue?

I mostly followed the style of the Xorg page for the position of everything. I will see if I can find a better location for Drivers.

Comment (ericonr) [reply src/config/wayland.md]: I moved the Video Drivers to below DEs and WMs, so I could refer to them without issue.

File: src/config/wayland.md (hunk 2 above)
Comment (Duncaen) [reply src/config/wayland.md]: `$XDG_RUNTIME_DIR` is usually on `/run` because only privileged users can create directories there opposed to `/tmp` where this could haven been maliciously created before your shell script runs on login.

Comment (ericonr) [reply src/config/wayland.md]: Considering this is a possible security hazard, do you think it should be documented as such, or fully removed?

Comment (flexibeast) [reply src/config/wayland.md]: i would prefer the documentation include a (more) secure way of setting up `XDG_RUNTIME_DIR`, since (a) it's necessary for Wayland, and (b) if people aren't provided with a more secure way, they'll instead use whatever 'solution' seems to 'work' - whether that's something they've put together themselves, or something they've just copied and pasted from somewhere else.

Comment (xtraeme) [reply src/config/wayland.md]: The right solution is to use elogind.

Comment (ericonr) [reply src/config/wayland.md]: All right, I will leave elogind as the only option. Do you think it's best to explain why elogind is necessary (it has to create the directory and set `XDG_RUNTIME_DIR`), or should I just mention that elogind is necessary for wayland and leave it at that?

Comment (ericonr) [reply src/config/wayland.md]: Fixed.

**Comment by bobertlo** [comment]
I really like the look of this. I have been waiting for the dust to settle because I am less savvy with Wayland. I will try to do a review and get this merged ASAP. Thanks for contributing!
